cmd/execute.go

```diff
@@ -305,21 +351,62 @@ func execute() error {
 	}()
 
 	// === Prometheus section ===
-	if metricsPort > 0 {
-		// Check if push or pull by looking at whether address is set or not
-		if metricsAddress == "" {
-			//Prometheus metrics will be pulled from the endpoint /metrics
-			err := buildmetrics.PullMetrics(ctx, metricsPort, scenario.RegisteredActions())
-			if err != nil {
-				return MetricError(fmt.Sprintf("failed to start prometheus : %s ", err))
+	if metricsLevel != "" || metricsPort > 0 {
+		var metricsType MetricLevel
+		var err error
+
+		// Temporary conversion from legacy code metrics arguments to new
+		if metricsPort > 0 {
```

Comment:
👍 would be good to also print a warning to console for usage of deprecated parameter so the user knows.